### PR TITLE
FlatGeobuf example notebook

### PR DESCRIPTION
**Pull request by kylebarron**
Closes https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/issues/13

### CODE REVIEW

File: flatgeobuf/flatgeobuf.ipynb [hunk 1]
```diff
@@ -0,0 +1,359 @@
+{
```
Comment (wildintellect) [inline flatgeobuf/flatgeobuf.ipynb:1]: <p>make the <code>fiona</code> paragraph a note/read more box.</p><p>I would rephrase our opinion - use <code>pyogrio</code> at the this time, and if you have new enough <code>gdal also use_arrow=True</code> (is this only for geoparquet?)  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (wildintellect) [inline flatgeobuf/flatgeobuf.ipynb:1]: <p>make a note box</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (wildintellect) [inline flatgeobuf/flatgeobuf.ipynb:1]: <p>Specify the order of variables in a comment (and if it's take a z?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (wildintellect) [inline flatgeobuf/flatgeobuf.ipynb:1]: <p>get the shape, not just the length</p><p>also are we going to add 1 more cell selecting only some columns?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (kylebarron) [reply flatgeobuf/flatgeobuf.ipynb:1]: I had assumed the special note box syntax didn't work from inside notebooks, but I'll move to using that.

Comment (kylebarron) [reply flatgeobuf/flatgeobuf.ipynb:1]: 99% of people who install pyogrio will do it via the wheels, not from source, in which case the bundled GDAL will be of version 3.6+. I should probably reword that version paragraph.

> is this only for geoparquet?

No, it's a bit confusing from the naming, but it works for every format. (This is related to [this comment](https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/25#discussion_r1297857919)). Originally pyogrio built the columnar arrays on its own in Cython, but now GDAL is able to build the columnar arrays _itself_ and pass those on to any environment that's able to understand Arrow arrays. So essentially the Arrow support _supplants_ the original pyogrio implementation, and will probably become the main method in the future because it's 50x less code (because it's all in the GDAL method)

**Comment by kylebarron** [comment]
I added another cell selecting only specific columns and this should be good to go now 🚀

File: flatgeobuf/flatgeobuf.ipynb [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
```
Comment (abarciauskas-bgse) [inline flatgeobuf/flatgeobuf.ipynb:1]: <p>Why do we need to download and save it? To emphasize reading from the cloud, I would skip this download and store step if at all possible.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (kylebarron) [reply flatgeobuf/flatgeobuf.ipynb:1]: Maybe it makes sense to reverse the order of the sections so that reading from the cloud happens before reading and writing locally?

I think I set it up this way because the cloud-native vector ecosystem is just younger and doesn't work as seamlessly as the raster/zarr sides. Like the remote fetch of vector features in my remote example took 2 minutes... I imagine some users would try to run that cell (on a slower internet connection than mine) and assume something went wrong.

Comment (abarciauskas-bgse) [reply flatgeobuf/flatgeobuf.ipynb:1]: I see 🤔 well if its possible to bypass downloading to get your points across, that would be preferable, but if not, perhaps we can just add a disclaimer similar to what you just wrote:

>The cloud-native vector ecosystem is younger and doesn't work as seamlessly as the COG and Zarr for subsetted access. We download data here to demonstrate important concepts but look to update these methods once better datasets and tools are available for working with flatgeobuf without downloading entire files.

**Review by abarciauskas-bgse — APPROVED** [review]
